### ISSUE TITLE
Categories are not working

### ISSUE THREAD

**Issue by asemiankevich** [dap-ps/discover#10]
View all link is not working (dev.dap.ps/discover-dapps/all)
Categories itself are not working at all -> can't see any dapp inside any category

![image](https://user-images.githubusercontent.com/28752642/60438982-938bb480-9c1a-11e9-98ca-c32b4c20c7c3.png)

**Comment by andytudhope** [comment]
This works well on my (release) version of Status?

**Comment by asemiankevich** [comment]
well i checked on Nightly and latest develop , maybe i was not clear in steps so here they are:

- open staus
- open dev.dap.ps
- tap view all
- try to drill into any category -> it is not tappable also not showing anything inside
- tap back button
- tap the http://dev.dap.ps/discover-dapps/all shortcut

**Comment by andytudhope** [comment]
It's not showing anything inside because there is currently nothing inside them. Trying to submit a new DApp using another category to test that the mapping is working, but not having any luck right now for some reason submitting at all.

They headers are not tappable either.

If I go

dev.dap.ps
tap back button
go to home screen

No problem?

**Comment by andytudhope** [comment]
What is much weirder is that some of the category mappings don't seem to work reliably. See this new test I did, called "Category Test".

1. Follow usual flow to submit DApp.
2. Select EXCHANGES as category and put in a description.
3. Submit and stake some SNT.
4. Approve DApp using postman.

See that the category I selected is not mapped correctly (and nothing appears under it), and that, in the profile page, the description is not working. @kstoykov do you have any idea why this might happen?

**Comment by kstoykov** [comment]
Hello, all the categories were hardcoded in the app. How did you created new category?

 

Best regards,

Kamen

**Comment by andytudhope** [comment]
EXCHANGES is one of the existing categories @kstoykov - we did not create a new one?

**Comment by kstoykov** [comment]
Oh, I got it now. I misunderstood you. I though you created a new category J.

 

So the dapp you created does not appear anywhere, right?

Usually you have to wait for transaction to finish, but I suppose you have done it.

Which server did you created the app on – locally? I mean can I see it somewhere? Also if you open the application with different browser do you see the dapp you created?

 

Best regards,

Kamen

**Comment by andytudhope** [comment]
No, it does not - apart from the Highest Ranked list. I did wait for the tx to be confirmed and have even approved the DApp using the API. It's created on dev.dap.ps - it seems EXCHANGE as a category does not work, as well as most of the others (though I haven't personally tested all of them).

**Comment by kstoykov** [comment]
I’ll check it and I’ll let you know as soon as possible.

**Comment by andytudhope** [comment]
@iamonuwa if you try to navigate directly to http://dev.dap.ps/discover-dapps/all, then the site does not load. This is an issue with React, do you have any ideas on how best to fix it so that any kind of weird route people choose, they'll still get to some part of the site?

**Comment by iamonuwa** [comment]
> @iamonuwa if you try to navigate directly to http://dev.dap.ps/discover-dapps/all, then the site does not load. This is an issue with React, do you have any ideas on how best to fix it so that any kind of weird route people choose, they'll still get to some part of the site?

See our conversation for this here https://github.com/status-im/discover-dapps/issues/31#issuecomment-484847833

**Comment by andytudhope** [comment]
It seems to me that, in this line, we just need to make sure that the ENV variable is not set to `development` and we should be ok, right @iamonuwa 

https://github.com/dap-ps/discover/blob/master/src/common/redux/store.js#L8

**Comment by iamonuwa** [comment]
If its hosted on GitHub pages, Yes

**Comment by iamonuwa** [comment]
> It seems to me that, in this line, we just need to make sure that the ENV variable is not set to `development` and we should be ok, right @iamonuwa
> 
> https://github.com/dap-ps/discover/blob/master/src/common/redux/store.js#L8

Correct.
